lib/rust/parser/generate-java/java/org/enso/syntax2/Parser.java

```diff
@@ -21,28 +22,28 @@ private Parser(long stateIn) {
     private static native void freeState(long state);
     private static native ByteBuffer parseInput(long state, ByteBuffer input);
     private static native long getLastInputBase(long state);
+    private static native long getMetadata(long state);
+    static native long getUuidHigh(long metadata, long codeOffset, long codeLength);
+    static native long getUuidLow(long metadata, long codeOffset, long codeLength);
```

Comment:
So far we needed fixed number of JVM->native calls per parsing request. I am afraid that with the `getUuidLow` and `getUuidHigh` this no longer is going to be the case! Will we need two calls per every `Tree`? Every `Tree` with UUID?

Why can't we serialize/deserialize UUID just like rest of the `Tree`?

Reply:
When I implement the new format (before I start frontend integration), everything will be done in Rust, and the Trees will have UUIDs as properties. This is just a quick solution to support the old format during the transitional period.